arbos/block_processor.go

```diff
@@ -320,6 +345,13 @@ func ProduceBlockAdvanced(
 				return nil, nil, err
 			}
 
+			if tx.Type() == types.ArbitrumInternalTxType {
+				state, err = arbosState.OpenSystemArbosState(statedb, nil, true)
+				if err != nil {
+					return nil, nil, err
+				}
+				header = createNewHeader(lastBlockHeader, l1Info, state, statedb, chainConfig, true)
```

Comment:
Let's do this outside of the closure. If opening the system ArbOS state fails, we want to fail the whole state transition, not just skip the rest of the internal transaction.

Reply:
done.